master/docs/conf.py

```diff
@@ -22,6 +22,17 @@
 
 # -- General configuration -----------------------------------------------------
 
+# Check if [docs] dependencies are installed
+try:
+    import sphinxcontrib.blockdiag
```

Comment:
This will raise `ImportError` that is not handled.
Also message is only printed to stdout, I suggest to raise exception instead.

I would do it in this way:

```python
try:
    import sphinxcontrib.blockdiag
except ImportError:
    raise RuntimeError("sphinxcontrib.blockdiag is not installed. Please install documentation dependencies with `pip install buildbot[docs]`")

import pkg_resources
try:
    pkg_resources.require('docutils>=0.8')
except resources.ResolutionError:
    raise RuntimeError("docutils is not installed or has incompatible version. Please install documentation dependencies with `pip install buildbot[docs]`")
```

Also, I'm worried that we will implicitly list all docs dependencies there, so it will not be true that "all documentation dependencies specified in the single place", which is one of the goals of this PR.